### PR TITLE
feat(tests): add test for large args offset with size zero on CALL opcode

### PR DESCRIPTION
**Pull request by JereSalo**
## 🗒️ Description

Hi! I believe a test like this exists for other opcodes but not for CALL, it is one of those that have a very large offset but size zero and therefore execution shouldn't revert because of the offset. In ethrex we had a bug that was fixed in [this PR](https://github.com/lambdaclass/ethrex/pull/3950). I created this test and tried it out before and after the fix and it detects the bug.
I am making these tests just for the CALL opcode but they apply to all xCALL opcodes, I don't know what the nicest way of making this more generic is. But generally this would be enough because these opcodes tend to have pretty similar logic inside of them.

## 🔗 Related Issues or PRs

N/A.

## ✅ Checklist

- [x] All: Ran fast `tox` checks to avoid unnecessary CI fails, see also [Code Standards](https://eest.ethereum.org/main/getting_started/code_standards/) and [Enabling Pre-commit Checks](https://eest.ethereum.org/main/dev/precommit/):
    ```console
    uvx --with=tox-uv tox -e lint,typecheck,spellcheck,markdownlint
    ```
- [x] All: PR title adheres to the [repo standard](https://eest.ethereum.org/main/getting_started/contributing/?h=contri#commit-messages-issue-and-pr-titles) - it will be used as the squash commit message and should start `type(scope):`.
- [x] All: Considered adding an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.
- [x] All: Set appropriate labels for the changes (only maintainers can apply labels).

### CODE REVIEW

File: tests/frontier/opcodes/test_call.py [hunk 1]
```diff
@@ -147,3 +147,54 @@ def test_call_memory_expands_on_early_revert(
...
+
+
+# TODO: There's an issue with gas definitions on forks previous to Berlin, remove this when fixed.
+# https://github.com/ethereum/execution-spec-tests/pull/1952#discussion_r2237634275
+@pytest.mark.valid_from("Berlin")
+def test_call_large_args_offset_size_zero(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    fork: Fork,
+):
```
Comment (LouisTsai-Csie) [inline tests/frontier/opcodes/test_call.py:159]: Thanks for this PR! If you want to expand the test to all the call related operation, you can parametrize these ops like:

```py
@pytest.mark.parametrize(
    "call_opcode",
    [
        Op.STATICCALL,
        Op.DELEGATECALL,
        Op.CALLCODE,
    ],
)
```

And then pass these opcode in the function, and replace the `Op.CALL` with `call_opcode` in your test.
```diff
@pytest.mark.valid_from("Berlin")
def test_call_large_args_offset_size_zero(
    state_test: StateTestFiller,
    pre: Alloc,
    fork: Fork,
+  call_opcode: Op
):
```

Some reference implementation [here](https://github.com/ethereum/execution-spec-tests/blob/6cf066e06b61d2a5ddd05104ba1274cabd5b6e5d/tests/osaka/eip7951_p256verify_precompiles/test_p256verify.py#L201) and [here](https://github.com/ethereum/execution-spec-tests/blob/6cf066e06b61d2a5ddd05104ba1274cabd5b6e5d/tests/osaka/eip7939_count_leading_zeros/test_count_leading_zeros.py#L622).

Comment (marioevz) [reply tests/frontier/opcodes/test_call.py:159]: I agree with this comment, and ideally we can use [`@pytest.mark.with_all_call_opcodes`](https://eest.ethereum.org/main/writing_tests/test_markers/#pytestmarkwith_all_call_opcodes) in this case.

Comment (JereSalo) [reply tests/frontier/opcodes/test_call.py]: You never cease to amaze me guys. Thank you very much.
Implemented changes [here](https://github.com/ethereum/execution-spec-tests/pull/2057/commits/656a47f01ce9e4401445ca47d4279cd073427fc3). Maybe in a follow-up I'll make changes to the other CALL tests in this file.

**Review by LouisTsai-Csie — COMMENTED** [review]
LGTM, let's wait for other members to review again and merge!
